### ISSUE TITLE
[Enhancement] Extend `opensearchDashboards.branding.applicationTitle` customization to app/opensearch_dashboards_overview

### ISSUE THREAD

**Issue by anubisg1** [opensearch-project/OpenSearch-Dashboards#3049]
**Describe the bug**

when configuring brainding, and specifically `opensearchDashboards.branding.applicationTitle` , the string is not applied to the page `app/opensearch_dashboards_overview`

**To Reproduce**

1. configure `opensearchDashboards.branding.applicationTitle: 'something'`
2. open dashbaords home page and navigate to "https://<dashboard-url/app/opensearch_dashboards_overview"
3. the title doesn't match the branding configuration

**Expected behavior**
Branding should be applied 

**OpenSearch Version**
2.4.0 

**Dashboards Version**
2.4.0

**Screenshots**

![image](https://user-images.githubusercontent.com/4518931/206424218-04469cda-b2b2-40d5-8914-60b3626f2c00.png)

**Comment by joshuarrrr** [comment]
@abbyhu2000 @kgcreative Is this intentional, or is this a string placement we missed in the original custom branding feature?

**Comment by kgcreative** [comment]
This feels like a missed string replacement.

**Comment by joshuarrrr** [comment]
Yeah, that tracks with the replacement areas highlighted in the original PR requirements. Anyone interested in fixing can also check out this PR for how specific title string replacements should work. https://github.com/opensearch-project/OpenSearch-Dashboards/pull/826

**Comment by abbyhu2000** [comment]
> @abbyhu2000 @kgcreative Is this intentional, or is this a string placement we missed in the original custom branding feature?

Yea it is intentionally excluded in the original custom branding project. When we drafted the original scope for custom branding, we discussed and decided to only include the 4 title strings that are highlighted in the PR description for title configuration for P0. (https://github.com/opensearch-project/OpenSearch-Dashboards/pull/826) We debated for a while on the specific occurrences of logos and titles in the application that should be allowed to customize, and came up with the list of 7 logos and 4 titles. The reason we decided to not include other title strings and logos for configuration is because we want to show that it is the wrapper around OpenSearch Dashboards that can be custom-branded, but we still want to keep the idea that OpenSearch Dashboards is the core functionalities, as for P0 of the project. 

Should we consider adding more branding configurations now? @joshuarrrr @kgcreative @kavilla

**Comment by joshuarrrr** [comment]
I don't think it's a matter of more configurations - just a matter of which strings can be customized with `applicationTitle` and which can't. Personally, I don't see much harm in extending the `applicationTitle` (given it's what @anubisg1 and other folks would prefer).

**Comment by joshuarrrr** [comment]
Updated the issue title and label to enhancement based on @abbyhu2000 's update.

**Comment by anubisg1** [comment]
> I don't think it's a matter of more configurations - just a matter of which strings can be customized with `applicationTitle` and which can't. Personally, I don't see much harm in extending the `applicationTitle` (given it's what @anubisg1 and other folks would prefer).

If i may, the reason why i reported this is because the actual home page has the application title changed, specifically here:

![image](https://user-images.githubusercontent.com/4518931/206989989-6d56a662-a072-4142-90b6-8234684cdbab.png)

when a user clicks on "visualize and Analyze" he/she's then redirected to the page in question... 
notice that the Icon remains branded, but not the application title.
As such, it looks like a miss (regardless if it was intentional or not) but more important it "breaks" the immerision provided by the branding and to a simple user (that has no idea of the configuration and what is supposed to change etc), it looks like a mistake

![image](https://user-images.githubusercontent.com/4518931/206990392-c1590107-7bc5-4eb1-ab65-d77c29c933dc.png)

**Comment by abbyhu2000** [comment]
@anubisg1  Thanks for call it out, and I can see how this might cause confusion for users. I also agree with @joshuarrrr that we can extend the title configuration to include the overview title. @kavilla What do you think?

**Comment by weily2** [comment]
![image](https://github.com/user-attachments/assets/4ccaa64e-c9be-4088-8f6e-0f44c635acfd)
![image](https://github.com/user-attachments/assets/dbc45579-641b-4562-8a79-48fa153e0610)
![image](https://github.com/user-attachments/assets/8555cfcb-92e1-494f-b273-4119c9731fd9)

The login page, overview page , side menu name will not be replaced after configure `opensearchDashboards.branding.applicationTitle: 'something'`

**Comment by weily2** [comment]
My opensearch-dashboards version is 2.15.0

**Comment by veryMidDev** [comment]
Is this issue solved in latest version or is there any workaround at the moment?
